File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
    const mySet = new Set(nums)
    let len=0 
    let max =0

    for (let i of mySet)
    {
        if (!mySet.has(i-1))
        {
            len = 0
            while (mySet.has(i +len))
                len +=1
            max = Math.max(max, len)
        }
    }
    return max
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mySet = set(nums)
        longe = 0

        for i in mySet:
            if (i-1) not in mySet:
                len = 1
                while (i+len) in mySet:
                    len = len + 1
                longe = max(len,longe)
        return longe